### PR TITLE
Metricbeat/RabbitMQ: Adds connections metricset

### PR DESCRIPTION
**Pull request by jaipradeesh**
Following metrics are added:
name - The name of the connection with non-ASCII characters escaped as in C.
vhost - Virtual host name with non-ASCII characters escaped as in C.
user - User name.
node - Node name.
channels - The number of channels on the connection.
channel_max - The maximum number of channels allowed on the connection.
frame_max - Maximum permissible size of a frame (in bytes) to negotiate with clients.
type - Type of the connection.
host - Server hostname obtained via reverse DNS, or its IP address if reverse DNS failed or was disabled.
peer_host - Peer hostname obtained via reverse DNS, or its IP address if reverse DNS failed or was not enabled.
peer_port - Peer port.
server_port - Server port.
packet_count - Number of packets sent, received and pending on the connection.
octet_count - Number of octets sent and received on the connection.

Signed-off-by: Jaipradeesh <jaipradeesh@gmail.com>

### CODE REVIEW

File: metricbeat/module/rabbitmq/connection/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      format: bytes
+    - name: type
+      type: keyword
+      description: >
+        Type of the connection.
+    - name: host
+      type: keyword
+      description: >
+        Server hostname obtained via reverse DNS, or its IP address if reverse DNS failed or was disabled.
+    - name: peer_host
```
Comment (ruflin) [inline metricbeat/module/rabbitmq/connection/_meta/fields.yml]: Can we make this `peer.host` and `peer.port?

File: metricbeat/module/rabbitmq/connection/connection.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+}
+
+type MetricSet struct {
+	mb.BaseMetricSet
+	*helper.HTTP
+}
+
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Experimental("The rabbitmq connection metricset is experimental")
```
Comment (ruflin) [inline metricbeat/module/rabbitmq/connection/connection.go]: Somehow docs and code do not align here. In the docs it's marked as beta. I assume that has to do with our generator :-( Could you change this to `cfgwarn.Beta`  here?

**Comment by ruflin** [comment]
@jaipradeesh One more thing, you could update the PR description with some more descriptive details?

**Comment by ruflin** [comment]
jenkins, test it

**Comment by ruflin** [comment]
To update the auto generated files you need to run `make update`.

**Comment by jaipradeesh** [comment]
- Hound's comments on avoid using `underscore` could be ignored? I followed similar pattern in other metricsets.
- Marked `connections` metricset as `beta`. Should I update `node` and `queue` metricsets (not changed in this PR) to `Beta` as well?
- Is there a make rule to populate `metricbeat/module/rabbitmq/connection/_meta
/data.json`. I've left it empty now.

@ruflin

**Comment by ruflin** [comment]
* For the `_` in variable names: Please don't use them. Sorry if we missed it in other PR's :-D
* Lets not touch the other metricsets in this PR. We can handle is in a separate discussion / PR.
* For the json output have a look at https://github.com/elastic/beats/blob/master/metricbeat/Makefile#L82 You can temporarely adjust the path that it runs only for your metricset
* For the failing CI: I would expect if you run `make update` and push all changes again, the CI should be resolved. I can tests it on my end if that does not help.

**Comment by jaipradeesh** [comment]
```
➜  metricbeat git:(metricbeat-rabbitmq-connections) ✗ make generate-json
mkdir -p /home/dolftax/code/src/github.com/jaipradeesh/beats/metricbeat/build
echo "BEAT_STRICT_PERMS=false" > /home/dolftax/code/src/github.com/jaipradeesh/beats/metricbeat/build/test.env
echo "ES_HOST="elasticsearch"" >> /home/dolftax/code/src/github.com/jaipradeesh/beats/metricbeat/build/test.env
echo "ES_PORT=9200" >> /home/dolftax/code/src/github.com/jaipradeesh/beats/metricbeat/build/test.env
echo "ES_USER=beats" >> /home/dolftax/code/src/github.com/jaipradeesh/beats/metricbeat/build/test.env
echo "ES_PASS=testing" >> /home/dolftax/code/src/github.com/jaipradeesh/beats/metricbeat/build/test.env
TESTING_ENVIRONMENT=snapshot-noxpack ES_BEATS=.. docker-compose -p metricbeatsnapshot-noxpack7.0.0-alpha1aca9c0ab94dae095b88dc234732debfcedff35ba  -f docker-compose.yml build 
Building zookeeper
Step 1/2 : FROM jplock/zookeeper:3.4.8
3.4.8: Pulling from jplock/zookeeper
4d06f2521e4f: Pull complete
a3ed95caeb02: Pull complete
8e87dca72717: Pull complete
8e4566461b6e: Pull complete
9c98c5e8ff98: Pull complete
Digest: sha256:de7732696dadb53a52c5f269e8812c5ca696c70849a30faf3a66b44943e6fb73
Status: Downloaded newer image for jplock/zookeeper:3.4.8
 ---> 80e2f8f23445
Step 2/2 : HEALTHCHECK --interval=1s --retries=90 CMD nc -z localhost 2181
 ---> Running in 64520792259a
Removing intermediate container 64520792259a
 ---> d09c06468ec8
..
..
..
---> Running in d4e7357cc2f6
Removing intermediate container d4e7357cc2f6
 ---> 4ed0d234c60e
Successfully built 4ed0d234c60e
Successfully tagged metricbeatsnapshotnoxpack700alpha1aca9c0ab94dae095b88dc234732debfcedff35ba_elasticsearch:latest
Building uwsgi_http
Step 1/8 : FROM python:3.6-alpine
 ---> 29b5ce58cfbc
Step 2/8 : RUN apk add --no-cache --virtual .build-deps gcc libc-dev linux-headers curl
 ---> Using cache
 ---> 4d59bc9624ca
Step 3/8 : RUN pip install --no-cache-dir --trusted-host pypi.python.org uwsgi
 ---> Using cache
 ---> f10497547069
Step 4/8 : WORKDIR /app
 ---> Using cache
 ---> 112222ba543b
Step 5/8 : COPY testdata/app /app
 ---> Using cache
 ---> cb12bcc2f3d5
Step 6/8 : HEALTHCHECK --interval=1s --retries=60 --timeout=10s CMD curl http://localhost:8080/
 ---> Using cache
 ---> 10a6f5ebf6c8
Step 7/8 : EXPOSE 8080 9191 9192
 ---> Using cache
 ---> d75537b62b36
Step 8/8 : CMD [""]
 ---> Using cache
 ---> c1762ccdacc4
Successfully built c1762ccdacc4
Successfully tagged metricbeatsnapshotnoxpack700alpha1aca9c0ab94dae095b88dc234732debfcedff35ba_uwsgi_http:latest
TESTING_ENVIRONMENT=snapshot-noxpack ES_BEATS=.. docker-compose -p metricbeatsnapshot-noxpack7.0.0-alpha1aca9c0ab94dae095b88dc234732debfcedff35ba  -f docker-compose.yml run beat go test -tags=integration github.com/elastic/beats/metricbeat/module/... -data
Creating network "metricbeatsnapshotnoxpack700alpha1aca9c0ab94dae095b88dc234732debfcedff35ba_default" with the default driver
warning: "github.com/elastic/beats/metricbeat/module/..." matched no packages
no packages to test
make: *** [Makefile:83: generate-json] Error 1
```

@ruflin Help.

**Comment by ruflin** [comment]
I think the problem is that you forked the project under `src/github.com/jaipradeesh/beats/metricbeat` in the go path instead of `src/github.com/elastic/beats/metricbeat`. I'm kind of surprised that you managed to run any tests at all for your module? Do you have a parallel fork under the correct directory?

**Comment by jaipradeesh** [comment]
Parallel fork. Fixing.

**Comment by ruflin** [comment]
I assume `make unit` works in the metricbeat directory?

**Comment by jaipradeesh** [comment]
https://github.com/elastic/beats/pull/6458/files#diff-8303434bf263f758175d97b66f919690R61 looks suspicious. Not sure which make rule added this. Debugging.

**Comment by ruflin** [comment]
There should not be a directory `kernel` in the `modules` directory. Can you remove the directory?

**Comment by jaipradeesh** [comment]
Removed. Generating JSON. It might take a while.

**Comment by ruflin** [comment]
You can temporarely adjust the line to
```
${DOCKER_COMPOSE} run beat go test -tags=integration github.com/elastic/beats/metricbeat/module/rabbitmq/connection/... -data
```

or even as following to gather data from a local rabbitmq instance you have running:
```
go test -tags=integration github.com/elastic/beats/metricbeat/module/rabbitmq/connection/... -data
```

**Comment by jaipradeesh** [comment]
@ruflin LGTM. Thank for helping with onboarding. Let me know.

**Comment by ruflin** [comment]
jenkins, test it

**Comment by ruflin** [comment]
@jaipradeesh Thanks for going through this. It's a pleasure to work like this.

**Pull request merged**